addons/udes_stock/models/stock_move.py

```diff
@@ -361,6 +361,24 @@ def group_by_key(self):
 
         return self.groupby(lambda ml: ml.u_grouping_key)
 
+    def _prepare_extra_values_for_group(self, pickings, moves):
+
+        values = {}
+
+        origins = list(set(pickings.mapped('origin')))
+        if len(origins) == 1:
+            values['origin'] = origins[0]
+
+        partners = pickings.mapped('partner_id')
+        if len(partners) == 1:
+            values['partner_id'] = partners.id
+
+        dates_done = list(set(moves.mapped('date')))
+        if len(dates_done) == 1:
+            values['date_done'] = dates_done[0]
```

Comment:
The above can be factored out as you're repeating the same logic 3x.

Reply:
It's not exactly the same logic, depends on the field type. the middle one uses.id and the other ones list(set()) and [0].
I will leave it like this, the next thing we add we factor, but I don't think we need anything else.